src/IB/IBFEMethod.cpp

```diff
@@ -651,7 +673,14 @@ IBFEMethod::preprocessIntegrateData(double current_time, double new_time, int nu
     // U^{n+1/2} and U^{n+1} to equal U^{n}.
     d_X_vecs->copy("solution", { "current", "new", "half" });
     d_U_vecs->copy("solution", { "current", "new", "half" });
-    d_F_vecs->copy("solution", { "half" });
+    switch (d_ib_solver->getTimeSteppingType())
+    {
+    case MIDPOINT_RULE:
+        d_F_vecs->copy("solution", { "current", "half" });
+        break;
+    default:
+        d_F_vecs->copy("solution", { "current", "new", "half" });
+    }
```

Comment:
This is a nice application of the copy syntax. I think I see why you wanted to use it immediately.